modules/codegen/src/main/scala/com/twilio/guardrail/generators/Scala/AkkaHttpServerGenerator.scala

```diff
@@ -166,40 +194,57 @@ object AkkaHttpServerGenerator {
         handlerName: String,
         methodSigs: List[scala.meta.Decl.Def],
         handlerDefinitions: List[scala.meta.Stat],
-        responseDefinitions: List[scala.meta.Defn]
+        responseDefinitions: List[scala.meta.Defn],
+        customExtraction: Boolean
     ) =
       for {
         _ <- Target.log.debug(s"renderHandler(${handlerName}, ${methodSigs}")
-      } yield q"""
-          trait ${Type.Name(handlerName)} {
+      } yield {
+        val tParams = if (customExtraction) List(customExtractionTypeParamContravariant) else List()
+        q"""
+          trait ${Type.Name(handlerName)}[..${tParams}] {
             ..${methodSigs ++ handlerDefinitions}
           }
         """
-    def getExtraRouteParams(tracing: Boolean) =
+      }
+    def getExtraRouteParams(customExtraction: Boolean, tracing: Boolean) =
       for {
         _ <- Target.log.debug(s"getExtraRouteParams(${tracing})")
-        res <- if (tracing) {
+        extractParam <- if (customExtraction) {
+          Target.pure(List(param"""customExtract: String => Directive1[$customExtractionTypeParamName]"""))
```

Comment:
There's one concern here -- If I supply `customExtract` with a `Directive1[Long] & Directive2[Boolean, String]` I'll end up with a `Directive3[Long, Boolean, String]`, which is not compatible with this type (expecting `Directive1`), effectively requiring `.tmap( (a,b,c) => Tuple1((a,b,c)) )` in order to align the types.

Attempting to reconcile this seems to require a little more thought, possibly `customExtract: String => Directive[E]`, which would force users to `new MyHandler[Tuple1[E]] { ... }`, but optimizes a bit more for composing `Directive`s.

What do you think?

Reply:
Hmm yeah very good point! Somehow I thought that your composition example would result in a `Directive1[(Long, Boolean, String)]`, but I just tried it, and you get back `Directive[(String, Int, Boolean)]`:

It does seem like `customExtract: String => Directive[E]` would be the way to go to enable that composition, which I think is important. It would be unexpected if `Directive1[Long] & Directive2[Boolean, String]` couldn't be passed in. The `new MyHandler[Tuple1[E]]` or is a little weird, but with some docs shouldn't be too bad?

Reply:
Yeah. Thinking about it a bit more, I really don't have a good way to work around this:
- **Option 1:** we could generate the unit special case, `new MyHandler_[Long]` where `abstract class MyHandler_[A] extends MyHandler[Tuple1[A]] { ... }`. **This is bad because** if someone tries to switch from `Directive1` to `Directive2`, they'll get an error message that drives them _away_ from simply removing the `_`.
- **Option 2:** we could provide a magnet "receiver" that lifts any `DirectiveN` to `Directive1[TupleN` via `.tmap(Tuple1(_))`, which would actually give us exactly the syntax we want, **at the expense of** magic that is more difficult to introspect, as well as introducing yet another thing that could go wrong.

Reply:
I think the easiest thing here is probably the best, just documenting the `Tuple1` requirement, as if a better solution presents itself in the future, it'll be backwards compatible with `Tuple1`, not forcing users to change anything in that case. If you need help with wording or examples, please feel free to reach out.

Reply:
Ah I see you already updated the documentation 😄 reviewing now